Andromeda customizations to support showing breaks/gaps in line charts

ewisor opened the pull request.

Commit:
Andromeda customizations to sup[port showing breaks/gaps in line charts.

Commit:
Set the default "No Data Value" to the largest negative float value (previous was the smallest positive float value - big difference!)

jmorales4 commented on MPChartLib/src/main/java/com/github/mikephil/charting/renderer/LineChartRenderer.java:
```diff
@@ -57,7 +57,7 @@
 
     // Andromeda Additions
-    private float noDataValue = Float.MIN_VALUE;
+    private float noDataValue = -Float.MAX_VALUE;
```
Yeah that is a big difference, and good catch!  makes me wonder about the other 5 places we use Float.MIN_VALUE in the code.  Questions: 
- Are we sure that -Float.MAX_VALUE is a valid float?  
- Is this flag a value that we encode into a protobuf?  If so, can the Acq produce exactly the same value
- Is using Float.NAN a better option for a noDataValue?

ewisor replied:
Are we sure that -Float.MAX_VALUE is a valid float? << Yes, I believe so

Is this flag a value that we encode into a protobuf? << I believe the ACQ side could set a float protobuf field to this value.  That said, I really selected this value to effectively disable the newly added feature.  For the cases where the chart is used, I think it is highly unlikely the ACQ would ever need to report (-Float.MAX_VALUE) as a valid value to be charted.

Is using Float.NAN a better option for a noDataValue? << It could work as a default value but only to disable the newly added feature, not as the actual value in a data set to indicate a gap/break should appear.

jmorales4 replied:
Yeah I agree with you that the ACQ would never need to send a valid data point with -Float.MAX_VALUE, I was more thinking about the ACQ needing to send it to indicate "No data"; what would that look like?

ewisor replied:
Gotcha now.  I looked into this for AutoTest pbuf results but punted since I also needed float fields to default to this value.  I opt'ed to just use same default value for both float and Int fields.  See TestResultsApi.proto  code below...
```
message Constants {
    // The primary "no result" value is zero.  This default works well for most metrics but not for a selected few.
    // For those exceptions we use the no_result_secondary value defined below.
    // This dual value approach is inconsistent but we only explicitly set defaults for the exceptions.
    optional int32 no_result_primary   = 10; // The primary default protobuffer value (zero)
    optional int32 no_result_secondary = 20
        [ default = -2147483648 ]; // Use as the default when zero is a valid metric value
    optional int32 result_less_than_one = 30
        [ default = -2147483647 ]; // Always use this when integer metric is less than one (e.g. <1ms)
}
message MinMaxAvgData {
    optional float min = 1 [ default = -2147483648.0 ];
    optional float max = 2 [ default = -2147483648.0 ];
    optional float avg = 3 [ default = -2147483648.0 ];
    optional float cur = 4 [ default = -2147483648.0 ];
}
```

Commit:
Changed default no data value to Float.NaN - this guarantees the newly added feature is disable by default.
Also, resolved a bug where check for the no data value was including values less than the no data value.  Now value must be exactly the same.
